### ISSUE TITLE
npm ERR! bcrypt@5.0.1 install: `node-pre-gyp install --fallback-to-build`

### ISSUE THREAD

**Issue by ved224555** [kelektiv/node.bcrypt.js#875]
npm ERR! code ELIFECYCLE
npm ERR! errno 9009
npm ERR! bcrypt@5.0.1 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 9009
npm ERR!
npm ERR! Failed at the bcrypt@5.0.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\sp479\AppData\Roaming\npm-cache\_logs\2021-05-17T17_30_38_939Z-debug.log

**Comment by rhythm98** [comment]
Hello, I got same error in node `v14.17.0` and `npm v7.13.0`  
```shell
npm ERR! command sh -c node-pre-gyp install --fallback-to-build
```

**Comment by vaishnaowankar** [comment]
Getting the same error. please provide a way to resolve asap.

**Comment by firebelley** [comment]
Just got this as well on a 64 bit raspberry pi. It appears the issue is that there is no `arm64` Linux build for release v5.0.1:
```
node-pre-gyp http GET https://github.com/kelektiv/node.bcrypt.js/releases/download/v5.0.1/bcrypt_lib-v5.0.1-napi-v3-linux-arm64-glibc.tar.gz
node-pre-gyp ERR! install response status 404 Not Found on https://github.com/kelektiv/node.bcrypt.js/releases/download/v5.0.1/bcrypt_lib-v5.0.1-napi-v3-linux-arm64-glibc.tar.gz
```

And indeed the 5.0.1 release assets don't list an arm64 version https://github.com/kelektiv/node.bcrypt.js/releases/tag/v5.0.1

Reverting to 5.0.0 fixes the issue in the meantime.

**Comment by rhythm98** [comment]
Try using [bcrypt.js](https://www.npmjs.com/package/bcryptjs)
```shell
$ npm i bcryptjs
```
```js
const bcrypt = require('bcryptjs')
// Auto-gen a salt and hash:
const hashedPassword = bcrypt.hashSync('yourPassword', 10);

// Load hash from your password DB.
bcrypt.compareSync("passwordToVerify", hash); // false
```

**Comment by ketu546** [comment]
> bcrypt@5.0.1 install D:\Document For move\Work Station\Online Classes\Authenticstion & Security\Secrets - Starting Code\node_modules\bcrypt
> node-pre-gyp install --fallback-to-build

The system cannot find the path specified.
internal/modules/cjs/loader.js:888
  throw err;
  ^

Error: Cannot find module 'D:\Document For move\Work Station\Online Classes\@mapbox\node-pre-gyp\bin\node-pre-gyp'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:885:15)
    at Function.Module._load (internal/modules/cjs/loader.js:730:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
npm WARN secrets@1.0.0 No description
npm WARN secrets@1.0.0 No repository field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! bcrypt@5.0.1 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the bcrypt@5.0.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\kiron\AppData\Roaming\npm-cache\_logs\2021-05-29T20_13_27_089Z-debug.log

**Comment by stanleyowen** [comment]
Hi, I recently face this problem too. Simply reinstall the `bcrypt` dependency using `npm` or `yarn` solves this problem for me.
```js
// Delete the bcrypt Dependency
npm uninstall bcrypt

// Download  bcrypt Dependency Again
npm install bcrypt
```

**Comment by tarnxxvr** [comment]
try this if not working ;-    bcrypt@5.0.1

**Comment by ShinChven** [comment]
Screw bcrypt, I use bcryptjs.

**Comment by GeorgeHester** [comment]
I know this is an old issue but I came across this same error today, I noticed that node-gyp was using node `v14.16.1`, for some reason nvm's default alias was set to that version, so I decided to update the the alias to the newest version of node.
`nvm alias default 18`
I then reinstalled bcrypt and it built without errors, seems to be that the script requires at least a certain version of node. Hope this helps.

**Comment by ShinChven** [comment]
> 

If you don't mind performance, I suggest you use `bcryptjs` over `bcrypt` to get rid of node-gyp problems.

**Comment by brandadriano** [comment]
> I know this is an old issue but I came across this same error today, I noticed that node-gyp was using node `v14.16.1`, for some reason nvm's default alias was set to that version, so I decided to update the the alias to the newest version of node. `nvm alias default 18` I then reinstalled bcrypt and it built without errors, seems to be that the script requires at least a certain version of node. Hope this helps.

how did you fix this? I mean, more details. sorry, it´s my first week on js and this error got me stuck :)

**Comment by LeydenJar** [comment]
For me it was the docker image I was using. When looking for the package on github, npm was trying to resolve using the system architecture, in this case resolving to https://github.com/kelektiv/node.bcrypt.js/releases/download/v5.0.1/bcrypt_lib-v5.0.1-napi-v3-linux-arm64-musl.tar.gz

Focus on linux-arm64..

If you look at the releases on the [repository](https://github.com/kelektiv/node.bcrypt.js/releases/tag/v5.0.1) you will see that this file doesn't exist.

Changing the docker image from node-alpine to node solved my problem
![image](https://user-images.githubusercontent.com/46806315/187011242-ee07cf65-4ff1-4679-81fb-7305d7588cbe.png)

**Comment by JHyeok** [comment]
This problem is fixed in bcrypt v5.1.0.
